File: Data Scientist/Modulo II/meus Scripts/Distribuicao_Binomial.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sat Oct 13 14:48:00 2018

@author: victortelles
"""

from scipy.stats import binom

#calculo da Probabilidade em uma distribuicao binomial
#jogar a moeda 5x e dar cara 3x

prob = binom.pmf(3, 5, 0.5)

#passar 4 sinais de 4 tempos cada qual a probabilidade de pegar 0, 1, 2, 3, 4 verdes

prob0 = binom.pmf(0, 4, 0.25)
prob1 = binom.pmf(1, 4, 0.25)
prob2 = binom.pmf(2, 4, 0.25)
prob3 = binom.pmf(3, 4, 0.25)
prob4 = binom.pmf(4, 4, 0.25)

probCum = binom.cdf(1, 4, 0.25) #Probabilidade cumulativa (até 1 sinal verde p(0) +p(1))

#passar 4 sinais de 2 tempos cada qual a probabilidade de pegar 0, 1, 2, 3, 4 verdes
prob0_2 = binom.pmf(0, 4, 0.5)
prob1_2 = binom.pmf(1, 4, 0.5)
prob2_2 = binom.pmf(2, 4, 0.5)
prob3_2 = binom.pmf(3, 4, 0.5)
prob4_2 = binom.pmf(4, 4, 0.5)

#Prova 12 Questoes - chutando todas as questoes, cada uma com 4 alternativas - qual a probabilidade de acertar 7

prob_prova = binom.pmf(7, 12, 0.25)

File: Data Scientist/Modulo II/meus Scripts/Distribuicao_Binomial.R
#calculo da Probabilidade em uma distribuicao binomial
#jogar a moeda 5x e dar cara 3x
dbinom(3, 5, 0.5)

#passar 4 sinais de 4 tempos cada qual a probabilidade de pegar 0, 1, 2, 3, 4 verdes
dbinom(0, 4, 0.25) #p(0) = 0.3164
dbinom(1, 4, 0.25) #p(1) = 0.4219
dbinom(2, 4, 0.25) #p(2) = 0.2109
dbinom(3, 4, 0.25) #p(3) = 0.0469
dbinom(4, 4, 0.25) #p(4) = 0.0039

pbinom(1, 4, 0.25) #Probabilidade cumulativa (at?? 1 sinal verde p(0) +p(1))

#passar 4 sinais de 2 tempos cada qual a probabilidade de pegar 0, 1, 2, 3, 4 verdes
dbinom(0, 4, 0.5) #p(0) = 0.0625
dbinom(1, 4, 0.5) #p(1) = 0.2500
dbinom(2, 4, 0.5) #p(2) = 0.3750
dbinom(3, 4, 0.5) #p(3) = 0.2500
dbinom(4, 4, 0.5) #p(4) = 0.0625

 
dbinom(7, 12, 0.25) #p(7) = 0.0115 = 1.15%





